### ISSUE TITLE
Action methods with async not triggering next Mutation Action. 

### ISSUE THREAD

**Issue by xmlking** [amcdnl/ngxs#14]
How to reproduce : https://stackblitz.com/edit/ngxs-simple-uchv5v
this works if we remove `async`

```ts
import { Store, Action, Mutation } from 'ngxs';

export class Add {}
export class AddSuccess {}

@Store({
  defaults: {
    count: 0
  }
})
export class AppStore {
  @Mutation(AddSuccess)
  addSuccess(state) {
    state.count++;
  }

 @Action(Add)
  async add(state) {
    return new AddSuccess()
  }

}
```

**Comment by xmlking** [comment]
workaround for my use case:  wrap the action function with `fromPromise`

```ts
import {Store, Action, Mutation} from 'ngxs';
import {OAuthService} from "angular-oauth2-oidc";
import {Login, LogoutSuccess, LoginSuccess, Logout} from "./auth.events";
import {fromPromise} from "rxjs/observable/fromPromise";

@Store({
  defaults: {
    isLoggedIn: false,
    profile: {}
  }
})
export class AuthStore {
  constructor(private oauthService: OAuthService) {
  }

  @Mutation(LoginSuccess)
  loginSuccess(state, {payload}) {
    state.isLoggedIn = payload.isLoggedIn;
    state.profile = payload.profile;
  }

  @Mutation(LogoutSuccess)
  logoutSuccess(state, {payload}) {
    state.isLoggedIn = false;
    state.profile = {};
  }

  @Action(Logout)
  logout(state, {payload}) {
    this.oauthService.logOut();
    return new LogoutSuccess()
  }

  //not working 
  @Action(Login)
  async login(state, { payload }) {
    // this.oauthService.initImplicitFlow();
    const profile = await this.oauthService.fetchTokenUsingPasswordFlowAndLoadUserProfile('sumodemo', 'sumodemo');
    const isLoggedIn = true;
    return new LoginSuccess({isLoggedIn, profile});
  }

  @Action(Login)
  workAround(state, { payload }) {
    return fromPromise( this.login(state, { payload }))
  }
}
```

**Comment by amcdnl** [comment]
Thats odd cuz it calls `forPromise` under the hood. See: https://github.com/amcdnl/ngxs/blob/master/src/ngxs.ts#L83

**Comment by deebloo** [comment]
@xmlking in your add method you don't have an await. Im not an expert with async functions but don't you need it to resolve or am I missing something?

**Comment by deebloo** [comment]
@amcdnl hmmmm im not sure if that is the case. to the internet!

**Comment by xmlking** [comment]
@amcdnl are you able to reproduce this issue ?

**Comment by abalad** [comment]
I'm having the same problem, I'm trying to fire a Mutation from within an action, but Mutation does not run.

```
import { Store, Mutation, Action } from 'ngxs';
import { AuthenticationService } from '../../shared/resources/authentication/authentication.service';
import { Login, LoginSuccess, LoginFailure } from '../actions/login.action';
import { LoginState } from '../reducers/login/login.state';
import { InitialState } from './login.state';

@Store({
  defaults: InitialState
})
export class LoginStore {

  constructor( private authenticationService: AuthenticationService ){}

  @Mutation(Login)
  load(state: LoginState, action: Login) {
    return {
      ...state,
      error: null,
      pending: true
    };
  }

  @Mutation(LoginSuccess)
  loginSuccess(state: LoginState, action: LoginSuccess) {
    return {
      ...state,
      error: null,
      pending: false
    };
  }

  @Mutation(LoginFailure)
  loginFailure(state: LoginState, action: LoginFailure) {
    console.log(state,  action);
    return {
      ...state,
      error: action.payload,
      pending: false
    };
  }

  @Action(Login)
  login(state, { payload }){
    return this.authenticationService.authenticate(payload)
      .then( (response) => { console.log(response); } )
      .catch( (error) =>  new LoginFailure(error.message)  ); // HE WILL NOT DISPATCH A MUTATION

  }

}
```

**Comment by amcdnl** [comment]
I wonder if its related to the error?

**Comment by amcdnl** [comment]
Side note: I'm gonna put in place a global error handling system

**Comment by abalad** [comment]
@amcdnl So it's similar, because my request `this.authenticationService.authenticate(payload) `also returns a promisse async

**Comment by perjansson** [comment]
I'm also seeing this problem where a Mutation is not called if returned from an async / await Action.

**Comment by amcdnl** [comment]
Fixed in 1.5.3
